### PR TITLE
chore(e2e-test): [Test automation] Verify Topology is a supported plugin and E2E scenario

### PR DESCRIPTION
**Pull request by teknaS47**
## Description

Added tests for the Topology plugin

This PR focuses on tests with most of the steps included in the issue [RHIDP-778](https://issues.redhat.com/browse/RHIDP-778) 

Will be creating another PR for for the remaining few tests.

[video.webm](https://github.com/user-attachments/assets/58115aeb-5b2d-4b06-9f63-3eb5df4952f0)

## Which issue(s) does this PR fix

- Fixes [RHIDP-778
](https://issues.redhat.com/browse/RHIDP-778)

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts [hunk 1]
```diff
@@ -15,11 +15,34 @@ test.describe('Test Topology Plugin', () => {
...
+      .locator('g')
+      .filter({ hasText: /Open URL/ })
+      .locator('image')
+      .first()
+      .click();
+    await page.getByLabel('Pod').click();
+    await page.getByLabel('Pod').getByText('1', { exact: true }).click();
+    await uiHelper.clickTab('Details');
+    await uiHelper.isTextVisible('Status');
+    await uiHelper.isTextVisible('Active');
```
Comment (subhashkhileri) [inline e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts:42]:
```suggestion
    await uiHelper.verifyText('Status');
    await uiHelper.verifyText('Active');
```
`isTextVisible` doesn't do assertion instead use `verifyText`

**Comment by invincibleJai** [comment]
Thanks @teknaS47,  good to update PR description with details on what does this PR covers along with GIF/videos demostrating the same as linked jira has 12 scenarios ?

File: e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts [hunk 2]
```diff
@@ -15,11 +15,48 @@ test.describe("Test Topology Plugin", () => {
     await common.loginAsGuest();
   });
 
-  test("Verify pods visibility in the Topology tab", async () => {
+  test("Verify pods visibility in the Topology tab", async ({ page }) => {
+    test.setTimeout(120000);
```
[the remark this answers is not in the document]
Comment (teknaS47) [reply e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts:19]: The test was failing due to a timeout a few commits ago, sometimes the pages take time to load and cause the test to fail on timeout, I added it to make sure it had enough time. Although it seems it is working fine now, I can change it back to default.

File: e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts [hunk 3]
```diff
@@ -15,11 +15,48 @@ test.describe("Test Topology Plugin", () => {
...
     await catalog.goToBackstageJanusProject();
     await uiHelper.clickTab("Topology");
     await uiHelper.verifyText("backstage-janus");
     await uiHelper.verifyText("rhdh");
     await uiHelper.verifyText("rhdh-rbac");
+    const url_button = page
```
[the remark this answers is not in the document]
Comment (teknaS47) [reply e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts]: Sorry, I missed your comment on this, but I made a few changes to the code now. Please let me know if some changes are required now. Thanks!

File: e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts [hunk 4]
```diff
@@ -15,11 +15,48 @@ test.describe("Test Topology Plugin", () => {
...
+    ).toBeVisible();
+    await expect(page.getByTitle("Deployment")).toBeVisible();
+    await uiHelper.verifyText("S");
+    await uiHelper.verifyText("RT");
+    await expect(
+      page.locator('[data-test-id="rhdh-rbac"] rect').first(),
+    ).toBeVisible();
+    await uiHelper.clickTab("Details");
+    await page.getByLabel("Pod").hover();
+    // await uiHelper.verifyText('1Running'); // To be added later
```
[the remark this answers is not in the document]
Comment (teknaS47) [reply e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts:51]: Those were just notes left for me as I was going to add those changes in the next PR. I will remove them for now.

File: e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (teknaS47) [reply e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts]: I get what you mean, I thought of it but decided not to do it. But yes I agree with your point, we should start following this pattern and make it easier for future edits. I will make the necessary changes.

File: e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts [hunk 5]
```diff
@@ -15,11 +15,48 @@ test.describe("Test Topology Plugin", () => {
...
+    await page.getByLabel("Display options").click();
+    await page.getByLabel("Pod count").click();
+    await uiHelper.verifyText("1");
+    await uiHelper.verifyText("Pod");
+    await page.getByLabel("Pod count").click();
+    // const code_decorator = page.getByRole('button', { name: 'Edit source code' }) // To be added later
```
[the remark this answers is not in the document]
Comment (teknaS47) [reply e2e-tests/playwright/e2e/plugins/topology/topology.spec.ts]: Removed the lines as it will be added in my next PR

**Comment by christoph-jerolimov** [comment]
I stopped the re-test to have CI bandwidth for PRs that try to solve the CI issues. Please wait with retests until we have a heartbeat again. Thanks for your patience.

/hold

**Pull request merged**